scenarios/test_codegen/test/lib_tests/SourceManager_test.res

```diff
@@ -818,7 +826,7 @@ describe("SourceManager fetchNext", () => {
 
     Assert.deepEqual(
       executeQueryMock.callIds,
-      ["0", "1", "4"],
+      ["2", "3", "4"],
       ~message="Should have skipped partitions that are at max queue size",
     )
```

Comment:
This test shows the best benefit of the new approach. See how partition 2 and 3 are the most behind. But previously we would fetch everything besides them, while now they are prioritized instead.